redoc.go

```diff
@@ -21,6 +23,7 @@ type Redoc struct {
 	SpecFS      *embed.FS
 	Title       string
 	Description string
+	Options     string
```

Comment:
Not a big fan of expecting a json string here.
It also seems that `json.Valid` doesn't ensure the string is an actual object.
https://go.dev/play/p/gmBO0XamJob

I see the redoc's Config type is quite [extensive](https://github.com/Redocly/redoc/blob/124309592670979d1a647718e5ad94b9b9e56e48/src/services/RedocNormalizedOptions.ts#L14) - so supporting all of these in a struct type might be overkill.

So why not use a `map[string]any` as type then later `json.Marshal` it?
That would ensure we're always receiving an object.

Your example would look like this:

```go
Options: map[string]any{
  "disableSearch": true,
  "theme": map[string]any{
    "colors": `{"primary":{"main":"#297b21"}}`,
  },
}
```

Reply:
Also instead of naming it `Options`, maybe we could name it `Config` or `Settings`? To be more similar to redoc's namings?

Reply:
@mvrilo the`map[string]any`is legit and make it stronger.

I have to upgrade to go 1.18 to handle the `any` type.

Reply:
@mvrilo if you have 5mn for the PR. It would be delightful.

Reply:
@mvrilo please.

Reply:
@mvrilo again pleasE.